### ISSUE TITLE
sdkmanrc zulu JVMs are missing in mise

### ISSUE THREAD

**Issue by rocketraman** [jdx/mise#1714]
**Describe the bug**

Version mapping of sdkman zulu JVMs seems to be faulty.

**To Reproduce**

My `.sdkmanrc` contains:

```
# Enable auto-env through the sdkman_auto_env config
# Add key=value pairs of SDKs to use below
java=11.0.22-zulu
```

which `sdk list java` shows as a valid JVM:

```
Zulu          |     | 21.0.2       | zulu    |            | 21.0.2-zulu
|     | 21.0.2.crac  | zulu    |            | 21.0.2.crac-zulu
|     | 21.0.2.fx    | zulu    |            | 21.0.2.fx-zulu
|     | 21.0.1       | zulu    |            | 21.0.1-zulu
|     | 21.0.1.crac  | zulu    |            | 21.0.1.crac-zulu
|     | 21.0.1.fx    | zulu    |            | 21.0.1.fx-zulu
|     | 17.0.10      | zulu    |            | 17.0.10-zulu
|     | 17.0.10.crac | zulu    |            | 17.0.10.crac-zulu
|     | 17.0.10.fx   | zulu    |            | 17.0.10.fx-zulu
|     | 17.0.9       | zulu    | installed  | 17.0.9-zulu
|     | 17.0.9.crac  | zulu    |            | 17.0.9.crac-zulu
|     | 17.0.9.fx    | zulu    |            | 17.0.9.fx-zulu
|     | 11.0.22      | zulu    |            | 11.0.22-zulu
|     | 11.0.22.fx   | zulu    |            | 11.0.22.fx-zulu
|     | 11.0.21      | zulu    |            | 11.0.21-zulu
|     | 11.0.21.fx   | zulu    |            | 11.0.21.fx-zulu
| >>> | 11.0.17      | zulu    | local only | 11.0.17-zulu
|     | 8.0.402      | zulu    |            | 8.0.402-zulu
|     | 8.0.402.fx   | zulu    |            | 8.0.402.fx-zulu
|     | 8.0.392      | zulu    |            | 8.0.392-zulu
|     | 8.0.392.fx   | zulu    |            | 8.0.392.fx-zulu
|     | 7.0.352      | zulu    |            | 7.0.352-zulu
|     | 6.0.119      | zulu    |            | 6.0.119-zulu
```

However, mise shows that JVM as missing:

```
# mise ls
Plugin  Version                 Config Source                               Requested
java    zulu-11.68.17
java    11.0.2
java    11.0.22-zulu (missing)  ~/source/foo/.sdkmanrc 11.0.22-zulu
java    17.0.2
java    21.0.1
node    10.24.1
node    16.20.2
node    20.11.0                 ~/.mise.toml                                20
```

and appears to have a completely different set of versions:

```
# mise ls-remote | grep zulu-11
zulu-11.2.3
zulu-11.29.3
zulu-11.29.11
zulu-11.31.11
zulu-11.33.15
zulu-11.35.13
zulu-11.35.15
zulu-11.37.17
zulu-11.39.15
zulu-11.41.23
zulu-11.43.21
zulu-11.43.55
zulu-11.45.27
zulu-11.48.21
zulu-11.50.19
zulu-11.52.13
zulu-11.54.23
zulu-11.54.25
zulu-11.56.19
zulu-11.58.15
zulu-11.58.23
zulu-11.60.19
zulu-11.62.17
zulu-11.64.19
zulu-11.66.15
zulu-11.66.19
zulu-11.68.17
```

**Expected behavior**
mise should be able to understand zulu JDK versions in `.sdkmanrc`.

**`mise doctor` output**

```text
version: 2024.2.18 linux-x64 (4893e09 2024-02-24)
activated: yes
shims_on_path: no

build_info: 
  Target: x86_64-unknown-linux-musl
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Sat, 24 Feb 2024 06:33:38 +0000
  Rust Version: rustc 1.76.0 (07dca489a 2024-02-04)
  Profile: release

shell: 
  /usr/bin/fish
  fish, version 3.7.0

dirs: 
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/.cache/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files: 
  ~/.config/mise/config.toml
  ~/.mise.toml
  ~/source/domains-registry-gateway/.sdkmanrc

plugins: 
  bun     (core)
  deno    (core)
  erlang  (core)
  go      (core)
  java    (core)
  node    (core)
  python  (core)
  ruby    (core)

toolset: 
  java@11.0.22-zulu  (missing)
  node@20.11.0       

env_vars: 
  MISE_SHELL=fish

settings: 
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cargo_binstall = true
  color = true
  disable_default_shorthands = false
  disable_tools = []
  experimental = true
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  node_compile = false
  not_found_auto_install = true
  paranoid = false
  plugin_autoupdate_last_check_duration = "7d"
  python_compile = false
  python_default_packages_file = "/home/raman/.default-python-packages"
  python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
  raw = false
  trusted_config_paths = []
  quiet = false
  verbose = false
  yes = false
  ci = false
  debug = false
  trace = false
  log_level = "info"
  python_venv_auto_create = false

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false

No problems found
```

**Additional context**

Judging from https://www.azul.com/downloads/?version=java-11-lts&package=jdk&show-old-builds=true#zulu, it appears that Zulu builds have an internal Azul version number (which Mise uses), and an OpenJDK version number (which sdkman uses). It appears this mismatch is the cause of the problem.

**Comment by jdx** [comment]
we use https://github.com/joschi/java-metadata which is likely where the issue lies

**Comment by rocketraman** [comment]
@jdx You should probably be using the `java_version` field rather than the `version` field.

**Comment by roele** [comment]
It seems that `asdf` and `sdkman` use a different versioning scheme for Java. While `asdf` uses `{vendor}-{vendor-version}` (e.g. zulu-11.68.17), `sdkman` seems to use `{java-version}-{vendor}` (e.g. 11.0.22-zulu). 

Since `mise` uses the `asdf` plugin ecosystem it is not supposed to be compatible with sdkman (@jdx correct me if i'm wrong here) and we should close this as "working as designed".

**Comment by rocketraman** [comment]
> Since `mise` uses the `asdf` plugin ecosystem it is not supposed to be compatible with sdkman (@jdx correct me if i'm wrong here) and we should close this as "working as designed".

@roele Mise currently has explicit support for `.sdkmanrc`, see https://github.com/jdx/mise/blob/main/src/plugins/core/java.rs#L343-L362.

So obviously it should either be compatible with it, OR it should remove support for it. I would then stop seeing errors from mise every time an `.sdkmanrc` is present.

**Comment by roele** [comment]
@rocketraman You're right, also found a reference here https://mise.jdx.dev/configuration.html#legacy-version-files.

**Comment by roele** [comment]
Played around with an approach to map the versions with #1719. Drawback would be that there are several unsupported vendors and zulu could only be mapped to the major version because of different versioning.

**Comment by roele** [comment]
@rocketraman Fix #1719 has been merged and released with 2024.3.1 which should improve this behaviour.

**Comment by rocketraman** [comment]
Thank you, mapping to the major version works for me and solves the problem.

```
$ mise ls
Plugin  Version        Config Source                Requested
java    zulu-11.68.17   ~/source/foo/.sdkmanrc      zulu-11
```
